src/pymodaq_utils/serialize/serializer.py

```diff
@@ -254,6 +254,104 @@ def deserialize(bytes_str: bytes) -> Tuple[List[SERIALIZABLE], bytes]:
         return list_obj, remaining_bytes
 
 
+class TupleSerializeDeserialize(SerializableBase):
+    @staticmethod
+    def serialize(tuple_object: Tuple) -> bytes:
+        """ Convert a tuple of objects into a bytes message together with the info to convert it back
+
+        Parameters
+        ----------
+        tuple_object: tuple
+            the tuple could contain whatever objects are registered in the SerializableFactory
+
+        Returns
+        -------
+        bytes: the total bytes message to serialize the tuple of objects
+
+        Notes
+        -----
+
+        The bytes sequence is constructed as:
+        * the length of the tuple
+
+        Then for each object:
+        * use the serialization method adapted to each object in the tuple
+        """
+        if not isinstance(tuple_object, tuple):
+            raise TypeError(f'{tuple_object} should be a tuple, not a {type(tuple_object)}')
+        return ListSerializeDeserialize().serialize(list(tuple_object))
+
+    @staticmethod
+    def deserialize(bytes_str: bytes) -> Tuple[Tuple[SERIALIZABLE], bytes]:
```

Comment:
```suggestion
    def deserialize(bytes_str: bytes) -> Tuple[Tuple[SERIALIZABLE, ...], bytes]:
```

A list can have (in typing) only a single type, therefore `list[str]` is a list of strings. Tuples, however, can have a determined length and different types, e.g. `tuple[str, int]` is a tuple of exactly one string and then exactly one int. So `tuple[str]` is a one-element tuple of one string. For more elements of the same type, you need `tuple[str, str, str]` (for three strings) or ` tuple[str, ...]` for undetermined number of strings.